File: src/main/java/me/miki/shindo/management/event/impl/EventLocationCape.java
package me.miki.shindo.management.event.impl;

import me.miki.shindo.management.event.Event;
import net.minecraft.client.network.NetworkPlayerInfo;
import net.minecraft.util.ResourceLocation;

public class EventLocationCape extends Event {

    private final NetworkPlayerInfo playerInfo;
    private ResourceLocation cape;

    public EventLocationCape(NetworkPlayerInfo playerInfo) {
        this.playerInfo = playerInfo;
    }

    public NetworkPlayerInfo getPlayerInfo() {
        return playerInfo;
    }

    public ResourceLocation getCape() {
        return cape;
    }

    public void setCape(ResourceLocation cape) {
        this.cape = cape;
    }
}

File: src/main/kotlin/me/miki/shindo/management/event/impl/EventLocationCape.kt
package me.miki.shindo.management.event.impl

import me.miki.shindo.management.event.Event
import net.minecraft.client.network.NetworkPlayerInfo
import net.minecraft.util.ResourceLocation

class EventLocationCape(
    private val _playerInfo: NetworkPlayerInfo,
) : Event() {
    private var _cape: ResourceLocation? = null

    fun getPlayerInfo(): NetworkPlayerInfo = _playerInfo

    fun getCape(): ResourceLocation? = _cape

    fun setCape(cape: Any?) {
        _cape = cape as? ResourceLocation
    }
}
